applications/GeoMechanicsApplication/custom_elements/triangle_2d_15.hpp

```diff
@@ -0,0 +1,1752 @@
+        typename BaseType::Pointer Create(const IndexType NewGeometryId,
+            PointsArrayType const& rThisPoints) const override
+        {
+            return typename BaseType::Pointer(new Triangle2D15(NewGeometryId, rThisPoints));
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /**
+         * @brief Creates a new geometry pointer
+         * @param rGeometry reference to an existing geometry
+         * @return Pointer to the new geometry
+         */
+        typename BaseType::Pointer Create(const BaseType& rGeometry) const override
+        {
+            auto p_geometry = typename BaseType::Pointer(new Triangle2D15(rGeometry.Points()));
+            p_geometry->SetData(rGeometry.GetData());
+            return p_geometry;
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /**
+         * @brief Creates a new geometry pointer
+         * @param NewGeometryId the ID of the new geometry
+         * @param rGeometry reference to an existing geometry
+         * @return Pointer to the new geometry
+         */
+        typename BaseType::Pointer Create(const IndexType NewGeometryId,
+            const BaseType& rGeometry) const override
+        {
+            auto p_geometry = typename BaseType::Pointer(new Triangle2D15(NewGeometryId, rGeometry.Points()));
+            p_geometry->SetData(rGeometry.GetData());
+            return p_geometry;
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /// Returns number of points per direction.
+        SizeType PointsNumberInDirection(IndexType LocalDirectionIndex) const override
+        {
+            if ((LocalDirectionIndex == 0) || (LocalDirectionIndex == 1)) {
+                return 5;
+            }
+            KRATOS_ERROR << "Possible direction index reaches from 0-1. Given direction index: "
+                << LocalDirectionIndex << std::endl;
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /**
+         * returns the local coordinates of all nodes of the current geometry
+         * @param rResult a Matrix object that will be overwritten by the result
+         * @return the local coordinates of all nodes
+         */
+        Matrix& PointsLocalCoordinates(Matrix& rResult) const override
+        {
+            rResult.resize(15, 2, false);
+            noalias(rResult) = ZeroMatrix(15, 2);
```

Comment:
actually if you're stting all the values this line can be avoided

Reply:
Done